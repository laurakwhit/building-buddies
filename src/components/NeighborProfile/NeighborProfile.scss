.neighbor-profile {
  display: flex;
  color: #112229;
  .main-content {
    overflow: scroll;
    width: calc(100% - 20rem);
    padding-top: 3rem;
    .neighbors-link {
      color: #112229;
      text-decoration: none;
      .fas {
        position: absolute;
        left: 24rem;
        font-size: 4rem;
        cursor: pointer;
        transition: ease-in-out 0.2s;
        &:hover {
          transform: scale(1.1);
          text-shadow: 2px 2px 2px rgba(0, 0, 0, 0.66);
        }
      }
    }
    .profile-image {
      width: 300px;
      height: 300px;
      border-radius: 50%;
    }
    .interests-container {
      color: #76939e;
    }
  }
}
