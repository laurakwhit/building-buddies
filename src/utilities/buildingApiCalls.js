export const getAllBuildings = async () => {
  const response = await fetch(`${process.env.REACT_APP_API}/api/v1/buildings`);
  const buildings = await response.json();
  return buildings;
};

export const getBuilding = async () => {};

export const getAllBuildingUsers = async buildingId => {
  const response = await fetch(
    `${process.env.REACT_APP_API}/api/v1/buildings/${buildingId}/users`
  );
  const buildingUsers = await response.json();
  return buildingUsers;
};
