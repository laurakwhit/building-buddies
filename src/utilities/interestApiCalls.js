export const getAllInterests = async () => {
  const response = await fetch(`${process.env.REACT_APP_API}/api/v1/interests`);
  const interests = await response.json();
  return interests;
};

export const addInterest = async () => {};
