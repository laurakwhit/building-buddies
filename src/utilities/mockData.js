export const mockInterests = [{ "id": 1, "name": "golf", "created_at": "2018-10-10T19:53:00.107Z", "updated_at": "2018-10-10T19:53:00.107Z" }, { "id": 2, "name": "gardening", "created_at": "2018-10-10T19:53:00.107Z", "updated_at": "2018-10-10T19:53:00.107Z" }, { "id": 3, "name": "camping", "created_at": "2018-10-10T19:53:00.107Z", "updated_at": "2018-10-10T19:53:00.107Z" }, { "id": 4, "name": "rock climbing", "created_at": "2018-10-10T19:53:00.107Z", "updated_at": "2018-10-10T19:53:00.107Z" }, { "id": 5, "name": "fly fishing", "created_at": "2018-10-10T19:53:00.107Z", "updated_at": "2018-10-10T19:53:00.107Z" }, { "id": 6, "name": "coding", "created_at": "2018-10-10T19:53:00.107Z", "updated_at": "2018-10-10T19:53:00.107Z" }, { "id": 7, "name": "working on cars", "created_at": "2018-10-10T19:53:00.107Z", "updated_at": "2018-10-10T19:53:00.107Z" }, { "id": 8, "name": "bowling", "created_at": "2018-10-10T19:53:00.107Z", "updated_at": "2018-10-10T19:53:00.107Z" }, { "id": 9, "name": "cooking", "created_at": "2018-10-10T19:53:00.107Z", "updated_at": "2018-10-10T19:53:00.107Z" }, { "id": 10, "name": "hiking", "created_at": "2018-10-10T19:53:00.107Z", "updated_at": "2018-10-10T19:53:00.107Z" }, { "id": 11, "name": "skiing", "created_at": "2018-10-10T19:53:00.107Z", "updated_at": "2018-10-10T19:53:00.107Z" }, { "id": 12, "name": "snowboarding", "created_at": "2018-10-10T19:53:00.107Z", "updated_at": "2018-10-10T19:53:00.107Z" }, { "id": 13, "name": "mountain biking", "created_at": "2018-10-10T19:53:00.107Z", "updated_at": "2018-10-10T19:53:00.107Z" }, { "id": 14, "name": "knitting", "created_at": "2018-10-10T19:53:00.107Z", "updated_at": "2018-10-10T19:53:00.107Z" }]
export const mockUser = {
  building_id: 291,
  email: "asdf@asdf.com",
  name: "asdf"
};