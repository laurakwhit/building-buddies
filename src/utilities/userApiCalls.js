export const getUserInterests = async user_id => {
  const response = await fetch(
    `${process.env.REACT_APP_API}/api/v1/users/${user_id}/interests`
  );
  const interests = await response.json();
  return interests;
};

export const addUser = async ({ name, email, password, building_id }) => {
  const response = await fetch(`${process.env.REACT_APP_API}/api/v1/users`, {
    method: 'POST',
    body: JSON.stringify({ name, email, password, building_id }),
    headers: {
      'Content-Type': 'application/json'
    }
  });
  const data = await response.json();
  return data;
};

export const addUserInterest = async ({ user_id, interest_id }) => {
  const response = await fetch(
    `${
      process.env.REACT_APP_API
    }/api/v1/users/${user_id}/interests/${interest_id}`,
    {
      method: 'POST'
    }
  );
  const data = await response.json();
  return data;
};

export const deleteUserInterest = async ({ user_id, interest_id }) => {
  const response = await fetch(
    `${
      process.env.REACT_APP_API
    }/api/v1/users/${user_id}/interests/${interest_id}`,
    {
      method: 'DELETE',
      body: JSON.stringify({ interest_id }),
      headers: {
        'Content-Type': 'application/json'
      }
    }
  );
  const data = await response.json();
  return data;
};
