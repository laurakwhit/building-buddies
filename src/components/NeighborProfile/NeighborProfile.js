import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Nav from '../Nav/Nav';
import { getUserInterests } from '../../utilities/userApiCalls';
import './NeighborProfile.scss';

class NeighborProfile extends Component {
  state = {
    neighborInterests: []
  };

  async componentDidMount() {
    const { neighbor } = this.props;
    const neighborInterests = await getUserInterests(neighbor.id);
    this.setState({ neighborInterests });
  }

  render() {
    const { neighbor } = this.props;
    const { neighborInterests } = this.state;
    let displayNeighborInterests;

    if (!neighborInterests.length) {
      displayNeighborInterests = (
        <p>This user does not have any interests yet!</p>
      );
    } else {
      displayNeighborInterests = neighborInterests.map(
        (neighborInterest, i) => (
          <p key={`${neighborInterest.id}-${i}`}>{neighborInterest.name}</p>
        )
      );
    }

    return (
      <div className="neighbor-profile">
        <Nav />
        <div className="main-content">
          <img
            src={require('../../assets/profile.jpg')}
            alt="profile"
            className="profile-image"
          />
          <h1>{neighbor.name} likes...</h1>
          <div className="interests-container">{displayNeighborInterests}</div>
        </div>
      </div>
    );
  }
}

NeighborProfile.propTypes = {
  neighbor: PropTypes.object.isRequired
};

export default NeighborProfile;
