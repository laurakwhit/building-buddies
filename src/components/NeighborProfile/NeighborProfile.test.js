import React from 'react';
import { shallow } from 'enzyme';

import NeighborProfile from './NeighborProfile';
import { mockNeighbor } from '../../utilities/mockData';

describe('NeighborProfile component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<NeighborProfile neighbor={mockNeighbor} />);
  });

  it('should match the snapshhot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
