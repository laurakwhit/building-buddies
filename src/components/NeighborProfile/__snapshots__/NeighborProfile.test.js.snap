// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NeighborProfile component should match the snapshhot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NeighborProfile
    neighbor={
      Object {
        "building_id": 1,
        "created_at": "2018-10-14T18:06:35.045Z",
        "email": "boolean.brandon@gmail.com",
        "id": 4,
        "name": "Boolean Brandon",
        "password": "asdfasdf",
        "updated_at": "2018-10-14T18:06:35.045Z",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Nav />,
        <div
          className="main-content"
        >
          <h1>
            Boolean Brandon
          </h1>
        </div>,
      ],
      "className": "neighbor-profile",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1>
            Boolean Brandon
          </h1>,
          "className": "main-content",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Boolean Brandon",
          },
          "ref": null,
          "rendered": "Boolean Brandon",
          "type": "h1",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Nav />,
          <div
            className="main-content"
          >
            <h1>
              Boolean Brandon
            </h1>
          </div>,
        ],
        "className": "neighbor-profile",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1>
              Boolean Brandon
            </h1>,
            "className": "main-content",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Boolean Brandon",
            },
            "ref": null,
            "rendered": "Boolean Brandon",
            "type": "h1",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
